docs/rules/new-regexp.md

```diff
@@ -0,0 +1,16 @@
+# Enforce the use of `new RegExp()` instead of `RegExp`
+
```

Comment:
Should we add a description here? Docs are quite... empty. But to be honest, does it need more content? It's quite straight forward now.